src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AbstractExpressionHandler.java

```diff
@@ -443,7 +443,7 @@ protected static DetailAST getFirstAstNode(DetailAST ast) {
                 curNode = curNode.getParent();
             }
             curNode = toVisit;
-        }
+        } while (curNode != null);
```

Comment:
This is same as before we need to avoid null at all if we can not, we should keep suppression as permanent

Reply:
We cannot, so lets just keep it as permanent.

There is no surviving mutations on this, so a lot of our tests says it has to be like this. The code is identical to our TreeWalker code. The code is looking for a node by line/column, and since we don't keep our nodes ordered in source order, doing a random scan is really the only way. I don't see any magical "find by..." ast here then we already have.

Reply:
Please copy this explanation to issue and let update sonar wantfix to "reason at #12680"

Reply:
> Please copy this explanation to issue 

Done.

> let update sonar wantfix to "reason at https://github.com/checkstyle/checkstyle/pull/12680"

Please do, I do not know how to update sonar.